libs/libc/stdlib/lib_rand48.c

```diff
@@ -0,0 +1,200 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <nuttx/config.h>
+
+#include <sys/types.h>
+#include <stdlib.h>
+
+#include <nuttx/lib/lib.h>
+
+/****************************************************************************
+ * Private Data
+ ****************************************************************************/
+
+static unsigned short g_seed48[7] =
+{
+  0,
+  0,
+  0,
+  0xe66d,
+  0xdeec,
+  0x5,
+  0xb
+};
+
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+
+#ifdef CONFIG_HAVE_LONG_LONG
+static uint64_t rand48_step(FAR unsigned short *xi, FAR unsigned short *lc)
+{
+  uint64_t a, x;
+
+  x = xi[0] | (xi[1] + (0ul << 16)) | (xi[2] + (0ull << 32));
+  a = lc[0] | (lc[1] + (0ul << 16)) | (lc[2] + (0ull << 32));
+  x = a * x + lc[3];
+
+  xi[0] = x;
+  xi[1] = x >> 16;
+  xi[2] = x >> 32;
+  return x & 0xffffffffffffull;
+}
+#endif
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: srand48
+ ****************************************************************************/
+
+void srand48(long seed)
+{
+  unsigned short p[3];
+
+  p[0] = 0x330e;
+  p[1] = seed;
+  p[2] = seed >> 16;
+  seed48(p);
+}
+
+/****************************************************************************
+ * Name: seed48
+ ****************************************************************************/
+
+FAR unsigned short *seed48(FAR unsigned short seed16v[3])
+{
+  static unsigned short p[3];
+
+  memcpy(p, g_seed48, sizeof p);
+  memcpy(g_seed48, seed16v, sizeof p);
+  return p;
+}
+
+/****************************************************************************
+ * Name: lcong48
+ ****************************************************************************/
+
+void lcong48(FAR unsigned short p[7])
+{
+  memcpy(g_seed48, p, sizeof(g_seed48));
+}
+
+/****************************************************************************
+ * Name: jrand48
+ *
+ * Description:
+ *   Return signed long integers uniformly distributed over the
+ *   interval [-2^31, 2^31).
+ *
+ ****************************************************************************/
+
+#ifdef CONFIG_HAVE_LONG_LONG
```

Comment:
why need guard jrand48/mrand48/nrand48/lrand48 by CONFIG_HAVE_LONG_LONG?

Reply:
Now it is because those are using rand48_step that is based on uint64_t. Maybe we can rework the code not to use uint64_t and remove the guard or have two variants depending on if long long is available or not.

Reply:
I think it should be possible to rework logic to to two integers instead of long long.

Reply:
Yes, you are right.